lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/SanitizeAndSplitBam.pm

```diff
@@ -0,0 +1,303 @@
+package Genome::InstrumentData::Command::Import::WorkFlow::SanitizeAndSplitBam;
+
+use strict;
+use warnings;
+
+use Genome;
+
+require IO::File;
+require List::MoreUtils;
+require List::Util;
+use Params::Validate ':types';
+
+class Genome::InstrumentData::Command::Import::WorkFlow::SanitizeAndSplitBam {
+    is => 'Command::V2',
+    roles => [qw/
+        Genome::InstrumentData::Command::Import::WorkFlow::Role::WithWorkingDirectory
+        Genome::InstrumentData::Command::Import::WorkFlow::Role::RemovesInputFiles
+    /],
+    has_input => [
+        bam_path => {
+            is => 'FilePath',
+            doc => 'The path of the unsorted bam to sort.',
+        },
+        library => {
+            is => 'Genome::Library',
+            doc => 'Library to use in the bam headers.',
+        },
+    ],
+    has_output => [ 
+        output_bam_paths => {
+            is => 'FilePath',
+            is_many => 1,
+            doc => 'The paths of the read group bams.',
+        },
+    ],
+    has_optional_transient => [
+        old_and_new_read_group_ids => { is => 'HASH', default => {}, },
+        _read_group_fhs => { is => 'HASH', default => {} },
+    ],
+};
+
+sub execute {
+    my $self = shift;
+    $self->debug_message('Spilt bam by read group...');
```

Comment:
These entry and exit debug messages could be updated to reflect the widened responsibility of this class.